### ISSUE TITLE
selenium.common.exceptions.SessionNotCreatedException: Message: session not created: DevToolsActivePort file doesn't exist.

### ISSUE THREAD

**Issue by gitkdp2844** [seleniumbase/SeleniumBase#2490]
![image](https://github.com/seleniumbase/SeleniumBase/assets/60216417/b75c0f63-f4a3-4e85-bf02-1bd99bb5c1e7)

same url is not loading

```
from selenium import webdriver
import time

#path = 'C:/Users/Pasupuleti.Chandra/Downloads/edgedriver/msedgedriver.exe'
#options = webdriver.EdgeOptions()
#options.add_argument('--headless')

driver = webdriver.Edge()
print("url opening============================================================================================================================")
driver.get('https://www.geeksforgeeks.com')
print("url opening============================================================================================================================")

time.sleep(540)
```

later i tried code as shown in picture.

```
from seleniumbase import Driver

driver = Driver(browser="edge")
try:
    driver.open("seleniumbase.io/simple/login")
    driver.type("#username", "demo_user")
    driver.type("#password", "secret_pass")
    driver.click('a:contains("Sign in")')
    driver.assert_exact_text("Welcome!", "h1")
    driver.assert_element("img#image1")
    driver.highlight("#image1")
    driver.click_link("Sign out")
    driver.assert_text("signed out", "#top_message")
finally:
    driver.quit()
```

** #NOTE: Dont close the issue untill confirmation**

**Comment by mdmintz** [comment]
What do you get when you try the example from Microsoft's website: https://learn.microsoft.com/en-us/microsoft-edge/webdriver-chromium/?tabs=python

```python
from selenium import webdriver
from selenium.webdriver.common.by import By
import time

driver = webdriver.Edge()

driver.get('https://bing.com')

element = driver.find_element(By.ID, 'sb_form_q')
element.send_keys('WebDriver')
element.submit()

time.sleep(5)
driver.quit()
```

Both that and the SeleniumBase example are working for me. Do you have the latest Edge and latest SeleniumBase?

**Comment by gitkdp2844** [comment]
![image](https://github.com/seleniumbase/SeleniumBase/assets/60216417/12a875f4-be63-44da-98e2-446a30043907)


The Bing url is not loading in browser. 
Yes!! i have selenium of version 4 and latest edge browser. 

selenium                  4.17.2
seleniumbase              4.23.3

![image](https://github.com/seleniumbase/SeleniumBase/assets/60216417/e79a417c-f14d-42aa-a4fb-0d6e6cfd9a8f)

**Comment by gitkdp2844** [comment]
[14152:17524:0213/093223.627:ERROR:policy_logger.cc(156)] :components\enterprise\browser\controller\chrome_browser_cloud_management_controller.cc(161) Cloud management controller initialization aborted as CBCM is not enabled. Please use the `--enable-chrome-browser-cloud-management` command line flag to enable it if you are not using the official Google Chrome build.

DevTools remote debugging is disallowed by the system admin.
[14152:17524:0213/093224.348:ERROR:edge_auth_errors.cc(516)] EDGE_IDENTITY: Get Default OS Account failed: Error: Primary Error: kImplicitSignInFailure, Secondary Error: kAccountProviderFetchError, Platform error: 0, Error string:

Traceback (most recent call last):
  File "c:\Users\Pasupuleti.Chandra\Desktop\test.py", line 5, in <module>
    driver = webdriver.Edge()
             ^^^^^^^^^^^^^^^^
  File "C:\Users\Pasupuleti.Chandra\AppData\Local\Programs\Python\Python311\Lib\site-packages\selenium\webdriver\edge\webdriver.py", line 45, in __init__
    super().__init__(
  File "C:\Users\Pasupuleti.Chandra\AppData\Local\Programs\Python\Python311\Lib\site-packages\selenium\webdriver\chromium\webdriver.py", line 61, in __init__
    super().__init__(command_executor=executor, options=options)
  File "C:\Users\Pasupuleti.Chandra\AppData\Local\Programs\Python\Python311\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 208, in __init__
    self.start_session(capabilities)
  File "C:\Users\Pasupuleti.Chandra\AppData\Local\Programs\Python\Python311\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 292, in start_session
    response = self.execute(Command.NEW_SESSION, caps)["value"]
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Pasupuleti.Chandra\AppData\Local\Programs\Python\Python311\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 347, in execute
    self.error_handler.check_response(response)
  File "C:\Users\Pasupuleti.Chandra\AppData\Local\Programs\Python\Python311\Lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 229, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.SessionNotCreatedException: Message: session not created: Microsoft Edge failed to start: exited normally.
  (session not created: DevToolsActivePort file doesn't exist)
  (The process started from msedge location C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe is no longer running, so msedgedriver is assuming that msedge has crashed.)
Stacktrace:
        GetHandleVerifier [0x00007FF777A0B892+60738]
        Microsoft::Applications::Events::ILogConfiguration::operator* [0x00007FF777997B32+233058]
        (No symbol) [0x00007FF7777AAE0B]
        (No symbol) [0x00007FF7777E305D]
        (No symbol) [0x00007FF7777DDB18]
        (No symbol) [0x00007FF777827389]
        (No symbol) [0x00007FF77781C233]
        (No symbol) [0x00007FF7777ED568]
        (No symbol) [0x00007FF7777EC394]
        (No symbol) [0x00007FF7777ED101]
        Microsoft::Applications::Events::EventProperty::to_string [0x00007FF777BC7D2C+1072924]
        (No symbol) [0x00007FF777870F45]
        (No symbol) [0x00007FF777908F6C]
        (No symbol) [0x00007FF777900A5D]
        Microsoft::Applications::Events::EventProperty::to_string [0x00007FF777BC6773+1067363]
        Microsoft::Applications::Events::ILogConfiguration::operator* [0x00007FF7779A1A88+273848]
        Microsoft::Applications::Events::ILogConfiguration::operator* [0x00007FF77799C794+252612]
        Microsoft::Applications::Events::ILogConfiguration::operator* [0x00007FF77799C8C2+252914]
        Microsoft::Applications::Events::ILogConfiguration::operator* [0x00007FF777992EC1+213489]
        BaseThreadInitThunk [0x00007FFDED6C257D+29]

**Comment by mdmintz** [comment]
It looks like your issue is related to having "Microsoft Edge for Business", rather than regular "Microsoft Edge".

After looking through Stack Overflow, I found a few things that might help. Here's an updated script:

```python
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.edge.options import Options
import time

options = Options()
options.add_argument("--remote-allow-origins=*")
options.add_argument("--enable-chrome-browser-cloud-management")
driver = webdriver.Edge(options=options)

driver.get('https://bing.com')

element = driver.find_element(By.ID, 'sb_form_q')
element.send_keys('WebDriver')
element.submit()

time.sleep(3)
driver.quit()
```

Although not directly related to SeleniumBase, fixing this will improve the situation for SeleniumBase users who have "Microsoft Edge for Business".

**Comment by gitkdp2844** [comment]
Nope again same thing nothing changed

**Comment by mdmintz** [comment]
OK, the above script uses raw Selenium, not SeleniumBase. I'm not sure I can help since it's not directly related to SeleniumBase and I don't have the "for Business" version of Edge.

I see something on Stack Overflow that might be useful: https://stackoverflow.com/questions/76017807/selenium-edge-browser-not-launching (If you figure out the options that fix the issue for you, I can add those options into SeleniumBase.)

It might also be a permissions issue since you had "DevTools remote debugging is disallowed by the system admin." I think if you have a System Administrator that you can check with, you might be able to get around the issue if they change your settings or something.

**Comment by mdmintz** [comment]
This message: `"DevTools remote debugging is disallowed by the system admin."` means that Selenium is being blocked by your System Administrator. Check with them about getting fewer restrictions so that you can run browser automation.

This is also indicated by your screenshot that says you are using "Microsoft Edge for Business". That's not the same as regular "Microsoft Edge".

And this is a duplicate of https://github.com/seleniumbase/SeleniumBase/issues/2489
